src/test/java/jenkins/branch/NoTriggerBranchPropertyTest.java

```diff
@@ -0,0 +1,107 @@
+
+    @Issue("JENKINS-32396")
+    @Test
+    public void smokes() throws Exception {
+        sampleRepo.init();
+        sampleRepo.git("checkout", "-b", "newfeature");
+        sampleRepo.write("stuff", "content");
+        sampleRepo.git("add", "stuff");
+        sampleRepo.git("commit", "--all", "--message=newfeature");
+        sampleRepo.git("checkout", "-b", "release", "master");
+        sampleRepo.write("server", "mycorp.com");
+        sampleRepo.git("add", "server");
+        sampleRepo.git("commit", "--all", "--message=release");
+        MultiBranchImpl stuff = r.jenkins.createProject(MultiBranchImpl.class, "stuff");
+        BranchSource branchSource = new BranchSource(new GitSCMSource(null, sampleRepo.toString(), "", "*", "", false));
+        branchSource.setStrategy(new NamedExceptionsBranchPropertyStrategy(new BranchProperty[0], new NamedExceptionsBranchPropertyStrategy.Named[] {
+            new NamedExceptionsBranchPropertyStrategy.Named("release*", new BranchProperty[] {new NoTriggerBranchProperty()})
+        }));
+        stuff.getSourcesList().add(branchSource);
+        r.configRoundtrip(stuff);
+        stuff.scheduleBuild2(0).getFuture().get();
+        r.waitUntilNoActivity();
+        showComputation(stuff);
+        FreeStyleProject master = r.jenkins.getItemByFullName("stuff/master", FreeStyleProject.class);
+        assertNotNull(master);
+        assertEquals(2, master.getNextBuildNumber());
+        FreeStyleProject release = r.jenkins.getItemByFullName("stuff/release", FreeStyleProject.class);
+        assertNotNull(release);
+        assertEquals(1, release.getNextBuildNumber());
+        FreeStyleProject newfeature = r.jenkins.getItemByFullName("stuff/newfeature", FreeStyleProject.class);
+        assertNotNull(newfeature);
+        assertEquals(2, newfeature.getNextBuildNumber());
+        sampleRepo.git("checkout", "master");
+        sampleRepo.write("file", "more");
+        sampleRepo.git("commit", "--all", "--message=master-2");
+        sampleRepo.git("checkout", "newfeature");
+        sampleRepo.write("file", "more");
+        sampleRepo.git("commit", "--all", "--message=newfeature-2");
+        sampleRepo.git("checkout", "release");
+        sampleRepo.write("file", "more");
+        sampleRepo.git("commit", "--all", "--message=release-2");
+        sampleRepo.notifyCommit(r);
+        showComputation(stuff);
+        assertEquals(3, master.getNextBuildNumber());
+        assertEquals(1, release.getNextBuildNumber());
+        assertEquals(3, newfeature.getNextBuildNumber());
+        QueueTaskFuture<FreeStyleBuild> releaseBuild = release.scheduleBuild2(0);
+        assertNotNull("was able to schedule a manual build of the release branch", releaseBuild);
+        assertEquals(1, releaseBuild.get().getNumber());
+        assertEquals(2, release.getNextBuildNumber());
+    }
+
+    static void showComputation(@NonNull ComputedFolder<?> d) throws Exception {
```

Comment:
Copied from `workflow-multibranch`. Could potentially be added to `ComputedFolder` itself, though it is useful only from tests.